src/ui/SimulationControls.tsx

```diff
@@ -0,0 +1,145 @@
+import { useContext, useState } from "react";
+import { CoreContext } from "../core/CoreContext";
+import { IAutomatonCore } from "../core/automatonCore";
+import { Kind, Mode } from "../core/core";
+
+export default function VisualWindows() {
+  const coreContext = useContext(CoreContext);
+  // used temporary as for state change signalization
+  const [automatonCore, setAutomatonCore] = useState<IAutomatonCore | null>(null);
+
+  const [simulationWord, setSimulationWord] = useState("aabc");
+  const [simulationWordPos, setSimulationWordPos] = useState(0);
+  const [wordRead, setWordRead] = useState<string[]>([]);
+  const [wordRemaining, setWordRemaining] = useState<string[]>([]);
+  const [steppping, setStepping] = useState(false);
+  const [stepInterval, setStepInterval] = useState<number>(1.0);
+
+  const startSimulation = () => {
+    if (coreContext == undefined) {
+      return;
+    }
+
+    const word = simulationWord.split("");
+
+    const core = coreContext.primary;
+    if (core.kind == Kind.AUTOMATON) {
+      core.runStart(word);
+      setAutomatonCore(core);
+      setSimulationWordPos(0);
+      setWordRemaining(word);
+      setWordRead([]);
+    }
+  };
+  const stopSimulation = () => {
+    if (coreContext == undefined) {
+      return;
+    }
+    const core = coreContext.primary;
+    if (core.kind == Kind.AUTOMATON) {
+      core.runEnd();
+      setAutomatonCore(null);
+      setSimulationWordPos(0);
+      setWordRemaining([]);
+      setWordRead([]);
+      stopTimer();
+    }
+  };
+
+  // returns wether succeded
+  function nextStep(): boolean {
+    if (automatonCore == null) {
+      return false;
+    }
+
+    const error = automatonCore.runNext();
+    if (error) {
+      if (error.details === "Input end reached") {
+        alert("Si na konci slova!");
+      } else {
+        console.log(error);
+      }
```

Comment:
```suggestion
      if (error.details === "Input end reached") {
        alert("Si na konci slova!");
      } if (error.details === "Step for the symbol doesn't exist in the delta function") {
        alert("Pre tento symbol neexistuje žiaden prechod!");
      } else {
        console.log(error);
      }
```
toto by mohlo byť lepšie zrozumiteľné pre používateľa

Reply:
dal som tam rovno vypis celej hlasky, aj ked je po anglicky